src/components/source_editor/BasisText.js

```diff
@@ -45,9 +45,13 @@ class BasisText extends React.Component {
                 this.setState({message: ''});
             }
         } catch (err) {
+            let message = "basis.validationError"
+            if (err.message === "Non-Periodic max atom limit exceeded") {
+                message = "basis.maxAtomError"
```

Comment:
We should adjust the `Made.parsers.xyz.validate` to also return the error code in a form of a number, perhaps:

- 0: when validation is successful
- 1001: when the basis is improperly structured
- 2001: when there are too many atoms

Then we should associate the error code with the corresponding `basis.<errorMessage>` through a config in a settings file (see examples in `enums.js` in MD or webapp).